src/native/minipal/types.h

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#ifndef HAVE_MINIPAL_TYPES_H
+#define HAVE_MINIPAL_TYPES_H
+
+#if defined(TARGET_32BIT) || defined(TARGET_OSX) || defined(TARGET_WINDOWS)
```

Comment:
cc @lambdageek we can move these https://github.com/dotnet/runtime/blob/844f6eff7f22b654e413c2c5491b1445f0d1ee13/src/mono/CMakeLists.txt#L774-L791 definitions here with a similar workaround for `HOST_` vs. `TARGET_` https://github.com/dotnet/runtime/blob/ec9fb02a2c6b606ef06acc911a0b104fd3d2a9a3/src/native/libs/System.IO.Compression.Native/extra_libs.cmake#L3

(I just realized that the macros in this file should be prefixed with `MINIPAL_` for consistency)